LanguageFeatures/Patterns/variable_A02_t02.dart

```diff
@@ -47,20 +47,58 @@ String testRecord(Record r, [bool doTypeTest = false]) {
   }
 }
 
-String testList(List l, [bool doTypeTest = false]) {
+String testRecord2(Record r) {
+  return switch (r) {
+    case (a, b) => "($a, $b)";
+    default => "default";
+  };
+}
+
+String testRecord3(Record r, [bool doTypeTest = false]) {
+  if (r case (a, b)) {
+    if (doTypeTest) {
+      a.expectStaticType<Exactly<int>>();
+      b.expectStaticType<Exactly<String>>();
+    }
+    return "($a, $b)";
+  } else {
+    return "default";
+  }
+}
+
+String testList1(List l, [bool doTypeTest = false]) {
   switch (l) {
     case [a, b]:
-      if (doTypeTest) {
-        a.expectStaticType<Exactly<int>>();
-        b.expectStaticType<Exactly<String>>();
-      }
-      return "[$a, $b]";
+    if (doTypeTest) {
+      a.expectStaticType<Exactly<int>>();
```

Comment:
Same issue as in LanguageFeatures/Patterns/variable_A02_t01.dart

Reply:
`dynamic` (I think the list has type `List<dynamic>`, but the github UI doesn't make it easy to check that right now).